client/login/magic-login/magic-login-email/magic-login-email-wrapper.tsx

```diff
@@ -0,0 +1,57 @@
+import { recordTracksEventWithClientId as recordTracksEvent } from 'calypso/state/analytics/actions';
+import { MagicLoginEmail } from '.';
+import './style.scss';
+
+interface MagicEmailDomainInfo {
+	domain: string;
+	name: string;
+	url: string;
+}
+interface MagicLoginEmailWrapperProps {
+	emailAddress: string;
+}
+const knownDomains: MagicEmailDomainInfo[] = [
+	{ domain: 'apple.com', name: 'Apple', url: 'https://www.icloud.com' },
+	{ domain: 'gmail.com', name: 'Gmail', url: 'https://mail.google.com' },
+	{
+		domain: 'outlook.com',
+		name: 'Outlook',
+		url: 'https://outlook.live.com',
+	},
+	{ domain: 'yahoo.com', name: 'Yahoo', url: 'https://mail.yahoo.com' },
+	{ domain: 'aol.com', name: 'AOL', url: 'https://mail.aol.com' },
+];
+
+export function MagicLoginEmailWrapper( { emailAddress }: MagicLoginEmailWrapperProps ) {
+	const getEmailDomain = ( email: string ): MagicEmailDomainInfo[] => {
+		const domainMatch = email.match( /@(.+)$/ );
+		const domain = domainMatch ? domainMatch[ 1 ].toLowerCase() : null;
+
+		const filteredDomains = knownDomains.filter( ( e ) => e.domain.toLowerCase() === domain );
+
+		// If no matches, we return the known domains.
+		return filteredDomains.length > 0 ? filteredDomains : knownDomains;
+	};
```

Comment:
You can export and use [this](https://github.com/Automattic/wp-calypso/blob/implement/login-by-site-url/packages/onboarding/src/utils/domain-suggester.ts#L17) function instead ✨

Reply:
Nice! Using it. Thank you.